sys/net/include/fib/fib.h

```diff
@@ -0,0 +1,141 @@
+ * @brief initializes all FIB entries with 0
+ */
+void fib_init(void);
+
+/**
+ * @brief de-initializes the FIB entries
+ */
+void fib_deinit(void);
+
+/**
+ * @brief Registration of reactive routing protocol handler function
+ */
+void fib_register_rrp(void);
+
+/**
+ * @brief Adds a new entry in the corresponding FIB table for global destination and next hop
+ *
+ * @param[in] iface_id       the interface ID
+ * @param[in] glob_dst       the destination address
+ * @param[in] glob_dst_size  the destination address size
+ * @param[in] next_hop       the next hop address to be updated
+ * @param[in] next_hop_size  the next hop address size
+ * @param[in] lifetime       the lifetime in ms to be updates
+ *
+ * @return FIB_SUCCESS on success
+ *         FIB_ERROR if the entry cannot be created due to unsufficient RAM
+ */
+int fib_table_add_entry(kernel_pid_t iface_id, uint8_t *glob_dst, size_t glob_dst_size,
+                        uint8_t *next_hop, size_t next_hop_size, uint32_t lifetime);
+
+/**
+ * @brief Updates an entry in the corresponding FIB table for global destination and next hop
+ *
+ * @param[in] iface_id       the interface ID
+ * @param[in] glob_dst       the destination address
+ * @param[in] glob_dst_size  the destination address size
+ * @param[in] next_hop       the next hop address to be updated
+ * @param[in] next_hop_size  the next hop address size
+ * @param[in] lifetime       the lifetime in ms to be updates
+ *
+ * @return FIB_SUCCESS on success
+ *         FIB_ERROR if the entry cannot be updated due to unsufficient RAM
+ */
+int fib_table_upd_entry(kernel_pid_t iface_id, uint8_t *glob_dst, size_t glob_dst_size,
+                        uint8_t *next_hop, size_t next_hop_size, uint32_t lifetime);
+
+/**
+ * @brief removes an entry from the corresponding FIB table
+ *
+ * @param[in] glob_dst       the destination address
+ * @param[in] glob_dst_size  the destination address size
+ */
+void fib_table_rem_entry(uint8_t *glob_dst, size_t glob_dst_size);
+
+/**
+ * @brief provides a next hop for a given destination
+ *
+ * @param[in] glob_dst            the destination address
+ * @param[in] glob_dst_size       the destination address size
+ * @param[in, out] iface_id       pointer to store the interface ID for the next hop
+ * @param[out] next_hop           pointer where the next hop address should be stored
+ * @param[in, out] next_hop_size  the next hop address size. The value is
+ *                                overwritten with the actual next hop size
+ *
+ * @return FIB_SUCCESS on success
+ *         FIB_ERROR_DESTINATION_UNREACHABLE if no next hop is available in any FIB table
+ *                                           all RRPs are notified before the return
+ *         FIB_ERROR_INSUFICCIENT_OUT_SIZE if the size for the next hop address is insufficient low
+ */
+int fib_get_next_hop(uint8_t *glob_dst, size_t glob_dst_size, kernel_pid_t *iface_id,
+                     uint8_t *next_hop, size_t *next_hop_size);
```

Comment:
> probably if the glob_dst is a full IPv6 address and the next_hop is a compressed IPv6 address. 

I'm still not sure how we would implement this behaviour, but sure for future references (or mesh-routing on link layer) this might be helpful. (since compressed IPv6 addresses without context [either the real context identifier from context-based compression or the surrounding link-layer packet for context-free compression] are basically useless).

> The used FIB entry type here provides type fields to store such additional information.

Wouldn't `ng_nettype_t` a fitting type for that?